FinancialTracker/ViewControllers/BalanceViewController.swift

```diff
@@ -92,20 +106,39 @@ class BalanceViewController: UIViewController {
         }
     }
     
-    @objc func signOut() {
-        authManager?.signOut { firebaseError, _ in
-            if let firebaseError = firebaseError {
-                switch firebaseError {
-                case .signOut(let error):
-                    guard let error = error else { return }
-                    self.present(UIAlertController.create(title: "Sign Out Error", message: error.localizedDescription), animated: true)
-                case .database, .unknown, .access, .auth, .nonExistingUser:
-                    assertionFailure("This error should not appear: \(firebaseError.localizedDescription)")
-                    // swiftlint:disable:next unneeded_break_in_switch
-                    break
+    @objc private func signOut() {
+        let alertController = UIAlertController(title: "Sign Out", message: "You are about to sign out. Are you sure?", preferredStyle: .alert)
+        alertController.addAction(UIAlertAction(title: "Confirm", style: .default, handler: { _ in
+            self.authManager?.signOut { authError, _ in
+                if let authError = authError {
+                    switch authError {
```

Comment:
Here you can optimise all these lines of code with an extension for the UIAlertController which takes say an AuthError in it's init and gives you back and alert based on that.